### PR TITLE
Update button minimum height on visionOS

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [ ] iOS
- [x] visionOS
- [ ] macOS

### Description of changes

As recommended by apple hig, buttons on visionOS should be at least 44 px tall. Until we have an official updated size ramp for visionOS, both `.small` and `.medium` buttons will have a minimum of 44.

### Binary change

Total increase: 1,048 bytes
Total decrease: 0 bytes
| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 31,515,176 bytes | 31,516,224 bytes | ⚠️ 1,048 bytes |
<details>
<summary> Full breakdown </summary>

| File | Before | After | Delta |
|------|-------:|------:|------:|
| ButtonTokenSet.o | 125,216 bytes | 126,264 bytes | ⚠️ 1,048 bytes |
</details>

### Verification

<details>
<summary>Visual Verification</summary>

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before | after |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/2015)

### CODE REVIEW

File: ios/FluentUI/Button/ButtonTokenSet.swift [hunk 1]
```diff
@@ -344,9 +344,9 @@ extension ButtonTokenSet {
             case .large:
                 return 52
             case .medium:
-                return 40
+                return Compatibility.isDeviceIdiomVision() ? 44 : 40
```
Comment (mischreiber) [inline ios/FluentUI/Button/ButtonTokenSet.swift:347]: How is this different from the iOS HIG, which also recommends 44pt high buttons on iOS?

Comment (mischreiber) [reply ios/FluentUI/Button/ButtonTokenSet.swift:347]: Actually, [per the HIG](https://developer.apple.com/design/human-interface-guidelines/buttons#Best-practices) even that's not quite right:
> As a general rule, a button needs a hit region of at least 44x44 pt — in visionOS, 60x60 pt — to ensure that people can select it easily, whether they use a fingertip, a pointer, their eyes, or a remote.

Comment (joannaquu) [reply ios/FluentUI/Button/ButtonTokenSet.swift:347]: that's hit area right? this is for the actual size of the button?

Comment (mischreiber) [reply ios/FluentUI/Button/ButtonTokenSet.swift:347]: Technically true, but our existing buttons all use the same hit and visual bounding boxes. And I'm not sure I'd want to change that except in specific one-off cases.

**Pull request closed without merging**